src/transpiling/mod.rs

```diff
@@ -150,11 +165,11 @@ impl ParsedSource {
     &self,
     transpile_options: &TranspileOptions,
     emit_options: &EmitOptions,
-  ) -> Result<EmittedSource> {
+  ) -> Result<EmittedSource, TranspileError> {
```

Comment:
I did this work as part of another PR to make a `transpile_owned(self)` method, but then ended up deciding to return the failure of `Arc::try_unwrap` in the `Ok` result to force the caller to handle it.